[Swap] Show address type for recipient

gromxyz opened the pull request:
Closes #1846

https://user-images.githubusercontent.com/87367763/137300334-06722323-d15d-4efa-bd80-1db938a8c7ba.mov

gromxyz committed:
Show walletType in swap

veado commented on src/renderer/components/swap/Swap.tsx:
```diff
@@ -1190,6 +1191,16 @@ export const Swap = ({
+      FP.pipe(
+        getWalletByAddress(oWalletBalances, targetWalletAddress),
+        O.map(({ walletType }) => walletType),
+        O.getOrElse(() => '')
```
Remove `O.getOrElse` and check `O.isSome(oMatchedWalletType)` in line 1287

veado commented on src/renderer/components/swap/Swap.tsx:
```diff
@@ -1271,7 +1282,10 @@ export const Swap = ({
-              <Styled.ValueTitle>{intl.formatMessage({ id: 'swap.recipient' })}</Styled.ValueTitle>
+              <div style={{ display: 'flex', flexDirection: 'row' }}>
+                <Styled.ValueTitle>{intl.formatMessage({ id: 'swap.recipient' })}</Styled.ValueTitle>
+                {matchedWalletType !== '' && <WalletTypeLabel>{matchedWalletType}</WalletTypeLabel>}
```
`O.isSome(oMatchedWalletType)` - see prev. comment.

veado commented on src/renderer/helpers/walletHelper.test.ts:
```diff
@@ -4,31 +4,45 @@ import * as NEA from 'fp-ts/lib/NonEmptyArray'
     asset: AssetRuneNative,
-    walletAddress: 'rune native wallet address'
+    walletAddress: 'rune native wallet address',
+    walletType: 'keystore' as WalletType
```
Try `const RUNE_WB: RUNE_WB` -> No need to use `as ...` for `WalletType`. Same for all other `XYZ_WB`

veado commented on src/renderer/helpers/walletHelper.ts:
```diff
@@ -112,3 +112,17 @@ export const addressFromWalletAddress = ({ address }: Pick<WalletAddress, 'addre
 ): O.Option<Address> => FP.pipe(oWalletAddress, O.map(addressFromWalletAddress))
+
+export const getWalletByAddress = (
+  oWalletBalances: O.Option<NonEmptyWalletBalances>,
```
Try to use `NonEmptyWalletBalances` w/o `O.Option` and change the logic in `matchedWalletType`

veado commented on src/renderer/helpers/walletHelper.ts:
```diff
@@ -112,3 +112,17 @@ export const addressFromWalletAddress = ({ address }: Pick<WalletAddress, 'addre
+
+export const getWalletByAddress = (
+  oWalletBalances: O.Option<NonEmptyWalletBalances>,
+  oAddress: O.Option<Address>
```
Same for `Address` - try w/o `Option` (see prev. comment)

veado reviewed (changes requested):
As mentioned in DM: Pls try to implement a more reactive way to change the label by entering/changing the address in the input (but not after clicking save icon).

gromxyz committed:
Make more reactive

veado commented:
After clicking cancel button, `wallettype` has not been updated.

## Example 1

- After canceling changes, `keystore` should be shown

https://user-images.githubusercontent.com/61792675/137341178-30ddb574-6590-4ab0-b26d-3b654c6242cd.mp4


## Example 2

- After entering `Ledger` address, but clicking cancel button, `keystore` should be shown

https://user-images.githubusercontent.com/61792675/137341196-02142f91-8991-4f8b-9056-29e096384b8e.mp4

gromxyz committed:
Fix wrong wallettype

veado committed:
Last tweaks

- Fix race condition in EditableAddress `inputOnKeyUpHandler`
- Tweak tests + styles
